### PR TITLE
Feature/jupyter

### PR DESCRIPTION
**Pull request by hellais**

### CODE REVIEW

**Comment by hellais** [comment]
@darkk Is it really the case that the inventory has a harcoded hash:

```
    "assertion": "groups.all | sort | join('|') | hash('sha256') == '2f901f46fcf5b1c09757db99e088d9c62a1148dfe70c0ad044a7100e5631fcdf'", 
```
How do I update it?

**Comment by darkk** [comment]
> How do I update it?

You verify that `./play inventory-check.yml` passes the check for all hosts in the inventory, and if it does, you update the hash.

It's there as an annoying protection against:
- hosts in inventory that don't have `dom0-bootstrap.yml` playbook applied to them
- dead hosts in inventory
- hosts placed to wrong DC in inventory

File: ansible/roles/jupyter/tasks/setup-anaconda.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+- name: install pkgs with pip
+  become: yes
+  become_user: root
+  command: '{{anaconda_link_dir}}/bin/pip install {{ item }}'
+  with_items: "{{ pip_pkgs }}"
+
+- name: remove conda-curl since it conflicts with the system curl
+  become: yes
+  become_user: root
```
Comment (darkk) [inline ansible/roles/jupyter/tasks/setup-anaconda.yml]: FYI: `become` and `become_user` are likely overridden by global variable `ansible_become: true`, that's mentioned in 1f8a9681331d07c95ab7ae3fd995406341068326 but I've lost link to corresponding ansible

File: ansible/roles/jupyter/templates/jupyter_notebook_config.py.j2 [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+#c.NotebookApp.open_browser = True
+
+## Hashed password to use for web authentication.
+#
+#  To generate, type in a python/IPython shell:
+#
+#    from notebook.auth import passwd; passwd()
+#
+#  The string should be of the form type:salt:hashed-password.
+c.NotebookApp.password = "{{ jupyter_password_hash }}"
```
Comment (darkk) [inline ansible/roles/jupyter/templates/jupyter_notebook_config.py.j2]: :+1:

**Comment by hellais** [comment]
I am going to merge this as it's deployed and travis is passing.

**Comment by darkk** [comment]
FYI, prometheus configuration was not updated, so `jupiter.ooni.io` metrics were not scraped.
I've done that running `./play deploy-prometheus.yml -t prometheus-conf`.

**Comment by hellais** [comment]
For my future self:

We may eventually want to use jupyterhub which seems like the thing you want for multi-user setups: http://jupyterhub.readthedocs.io/en/latest/

**Pull request merged**